### DISCUSSION TITLE
chart define and yaml variables

### DISCUSSION THREAD

**Discussion by nrmn2492** [argoproj/argo-cd#8727] [General]
👋 
I am new to helm-argocd deployment, and I would have 2 questions that arose when using "agocd".

1.
Do I have any options to define the chart and helm values files stored in a separate path?
e.g.

apiVersion: argoproj.io/v1alpha1
kind: Application
> ....
>     helm:
>       valueFiles:
>        - Value-Stack/2/values.yaml
>       chartFile: (??)
>        - Chart-Stack/2/
>       version: v3
>     path: k8s-helm/nodejs-app
>     repoURL: https://foobar@bitbucket.org/foo/asd.git
> ...
---------------
2.
and is there a way to pass yaml values with environment variables?
e.g.

argocd-create.yaml:

> apiVersion: argoproj.io/v1alpha1
> kind: Application
> metadata:
>   name: $NODE_NAME
> spec:
>   destination:
>     namespace: default
>     server: https://kubernetes.default.svc
>   project: default
>   source:
>     helm:
>       valueFiles:
>        - $VALUE_FILE
>       version: v3
>     path: $REPO_PATH
>     repoURL: https://foobar@bitbucket.org/foo/asd.git

   
  
`argocd app create node3 -f argocd-create.yaml -p $NODE_NAME=node3 $VALUE_FILE=values/values.yaml $REPO_PATH=k8s-helm`

**Comment by todaywasawesome** [comment]
Yes and yes. See the docs here https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#values-files
 `argocd app set helm-guestbook --values values-production.yaml`

> Values files must be in the same git repository as the Helm chart. The files can be in a different location in which case it can be accessed using a relative path relative to the root directory of the Helm chart.

And

`argocd app set helm-guestbook -p service.type=LoadBalancer`

> Similarly, Argo CD can override values in the values.yaml parameters using argo app set command, in the form of -p PARAM=VALUE. For example:
